Unwrap multiple targets

ruggi opened the pull request:
Fixes #4018 

**Problem:**

- Unwrapping multiple groups with CMD+G only selects the children of the first unwrapped group
- Unwrapping multiple groups with the context menu only unwraps one of the groups, as multiple unwrapping is not supported

![Kapture 2023-08-01 at 16 38 36](https://github.com/concrete-utopia/utopia/assets/1081051/f72b994f-dd7e-4a4c-9a26-a35b8094ead6)

This is because:
1. The unwrapping for multiple selected views works by enqueueing sequential unwraps, each of which resets the selection
2. The unwrapping for the context menu triggers the unwrap on the `0th` element of the selection

**Fix:**

- Rework the unwrapping logic so that it supports multiple targets, which solves both the above in one go
- Make sure the old tests pass
- Add specific tests

![Kapture 2023-08-01 at 16 39 41](https://github.com/concrete-utopia/utopia/assets/1081051/20e95336-3246-47d3-951e-668adc787ef6)

**Notes:**

I tried to touch the logic of the unwrapping as little as possible: there's quite a bit of room for improvement there, but imho this is already a relevant enough change that it should simply make sure the existing logic works fine as-is, with just the minimum set of additions to work with the multiple targets approach, without adding all the cognitive load of a larger rewrite.

seanparsons commented on editor/src/components/context-menu-items.ts:
```diff
@@ -402,9 +402,7 @@ export const unwrap: ContextMenuItem<CanvasData> = {
-    if (data.selectedViews.length > 0) {
-      requireDispatch(dispatch)([EditorActions.unwrapElement(data.selectedViews[0])], 'everyone')
-    }
+    requireDispatch(dispatch)([EditorActions.unwrapElements(data.selectedViews)], 'everyone')
```
Lovely!

seanparsons commented on editor/src/components/editor/actions/actions.spec.browser2.tsx:
```diff
@@ -5873,3 +6102,7 @@ export var storyboard = (
   })
 })
+
+function comparePaths(a: ElementPath, b: ElementPath): number {
```
We should probably put this in `element-path.ts`.

seanparsons commented:
Does do the right thing if we had something like:
```
Group 1 (selected)
+ Div 1
  + Group 2 (selected)
    + Div 2
```
Then unwrap things which I'd expect to result in:
```
Div 1
+ Div 2
```

ruggi commented:
> Does this do the right thing if we had something like:
> 
> ```
> Group 1 (selected)
> + Div 1
>   + Group 2 (selected)
>     + Div 2
> ```
> 
> Then unwrap things which I'd expect to result in:
> 
> ```
> Div 1
> + Div 2
> ```

_almost_, it was not correctly cleaning up the nested group. I adjusted and added a specific test for it here https://github.com/concrete-utopia/utopia/pull/4019/commits/f68e730acf13bee7e401bdb1dedc508443ffa7e1
